File: tugas2/queue1/main.py
from collections import deque

class AntrianArray:
    def __init__(self, ukuran):
        self.data = [None] * ukuran
        self.depan = 0
        self.belakang = 0
        self.maks_elemen = ukuran

    def insert(self, x):
        posisi_belakang = (self.belakang + 1) % self.maks_elemen

        if posisi_belakang == self.depan:
            print("Antrian Penuh")
            print(f"{x} tidak dimasukkan")
        else:
            self.belakang = posisi_belakang
            self.data[self.belakang] = x

    def remove(self):
        if self.empty():
            print("Antrian kosong")
            return ""
        self.depan = (self.depan + 1) % self.maks_elemen
        return self.data[self.depan]

    def empty(self):
        return self.depan == self.belakang

def antrian_deque():
    daftar = deque()
    daftar.extend([
        "Aminudin",
        "Bahtiar Rivai",
        "Cita Citata",
        "Dul Sumbang",
        "Engel Lelga",
        "Farah Nurlaela",
        "Galih",
        "Herwantyoko",
        "Indah Kusuma"
    ])

    print("\n[Deque] Isi Antrian:")
    while daftar:
        nama = daftar.popleft()
        print(nama)

def main():
    print("Isi Antrian:")
    ukuran = 8
    daftar_array = AntrianArray(ukuran)

    nama_nama = [
        "Aminudin",
        "Bahtiar Rivai",
        "Cita Citata",
        "Dul Sumbang",
        "Engel Lelga",
        "Farah Nurlaela",
        "Galih",
        "Herwantyoko",
        "Indah Kusuma"
    ]

    for nama in nama_nama:
        daftar_array.insert(nama)

    while not daftar_array.empty():
        print(daftar_array.remove())

    antrian_deque()


if __name__ == "__main__":
    main()

File: tugas2/queue1/main.java
public class antrian1 {
    public static void main(string[] args){
        int ukuran = 8;
        Antrian daftar = new Antrian(ukuran);

        daftar.insert("Aminudin");
        daftar.insert("Bahtiar Rivai");
        daftar.insert("Cita Citata");
        daftar.insert("Dul Sumbang");
        daftar.insert("Engel Lelga");
        daftar.insert("Farah Nurlaela");
        daftar.insert("Galih");
        daftar.insert("Herwantyoko");
        daftar.insert("Indah Kusuma");

        System.out.println('Isi Antrian:');
        while(!daftar.empty()){
            String nama = daftar.remove();
            System.out.println(nama);
        }
    }
}

class Antrian {
    private String[] data;
    private int depan, belakang;
    private int maksElemen;

    public Antrian(int ukuran){
        depan = 0;
        belakang = 0;
        maksElemen = 0;
        data = new String[ukuran];
    }

    public void insert(String x){
        int posisiBelakang;

        if(belakang == maksElemen -1){
            posisiBelakang = 0;
        } else{
            posisiBelakang = belakang+1;
        }

        if(posisiBelakang == depan){
            System.out.println("Antrian Penuh");
            System.out.println(x + "tidak dimasukkan");
        }else{
            belakang = posisiBelakang;
            data[belakang] = x;
        }
    }

    public String remove(){
        if(empty()){
            System.put.println("Antrian kosong");
            return "";
        }
        if(depan == maksElemen-1){
            depan = 0;
        }else{
            depan=depan+1;
            return data[depan];
        }
    }

    public boolean empty(){
        if(depan==belakang){
            return true;
        }else{
            return false;
        }
    }
}